completion_aggregator/tasks.py

```diff
@@ -4,36 +4,78 @@
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 
+import logging
 from collections import namedtuple
 from datetime import datetime
 
 import pytz
 from celery import shared_task
+from opaque_keys.edx.keys import CourseKey, UsageKey
 from xblock.completable import XBlockCompletionMode
 from xblock.core import XBlock
 
-from django.conf import settings
+from django.contrib.auth.models import User
 
+from . import compat
 from .models import Aggregator
 
 OLD_DATETIME = pytz.utc.localize(datetime(1900, 1, 1, 0, 0, 0))
 
+log = logging.getLogger(__name__)
 
-CompletionStats = namedtuple('CompletionStats', ['earned', 'possible', 'last_modified'])  # pylint: disable=invalid-name
+
+CompletionStats = namedtuple('CompletionStats', ['earned', 'possible', 'last_modified'])
 
 
 @shared_task
-def update_aggregators(user, course_key, block_keys=frozenset()):  # pylint: disable=unused-argument
+def update_aggregators(username, course_key, block_keys=(), force=False):
     """
-    Update aggregators for the specified course.
+    Update aggregators for the specified enrollment (user + course).
+
+    Parameters
+    ----------
+        username (str):
+            The user whose aggregators need updating.
+        course_key (str):
+            The course in which the aggregators need updating.
+        block_key (list[str]):
+            A list of completable blocks that have changed.
+        force (bool):
+            If True, update aggregators even if they are up-to-date.
+
+    Takes a collection of block_keys that have been updated, to enable future
+    optimizations in how aggregators are recalculated.
 
-    Takes a collection of block_keys that have been updated, to enable
-    future optimizations in how aggregators are recalculated.
     """
-    from xmodule.modulestore.django import modulestore   # pylint: disable=import-error
+    user = User.objects.get(username=username)
+    course_key = CourseKey.from_string(course_key)
+    block_keys = set(UsageKey.from_string(key) for key in block_keys)
+    return _update_aggregators(user, course_key, block_keys, force)
```

Comment:
The main difficulty I had when debugging all of this was that when celery tasks fail from uncaught exceptions, they don't log anything by default; the tasks just silently die.  So I wrapped this call in a try/catch, and logged the exceptions to diagnose the issues noted below.

The celery-standard way to do this is to subclass `celery.Task`, override the `on_failure` method, and log the exception there.  See edx-platform's [instructor task](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/instructor_task/tasks_base.py) for an example, cf https://stackoverflow.com/a/45333231.

I'd recommend doing that now, so that when you adjust the conditions for updating aggregators, you've got a better chance of catching when things go wrong.